marklogic-client-api/src/main/java/com/marklogic/client/impl/OkHttpServices.java

```diff
@@ -3569,8 +3570,21 @@ public void postBulkDocuments(
           mimeType = defaultFormat.getDefaultMimetype();
         }
         headers.add(HEADER_CONTENT_TYPE, mimeType);
-        String disposition = DISPOSITION_TYPE_ATTACHMENT + "; " +
-          DISPOSITION_PARAM_FILENAME + "=" + escapeContentDispositionFilename(write.getUri()) + contentDispositionTemporal;
+        String disposition = null;
+        CharsetEncoder asciiEncoder = java.nio.charset.StandardCharsets.US_ASCII.newEncoder();
+        if(asciiEncoder.canEncode(write.getUri())) {
+            disposition = DISPOSITION_TYPE_ATTACHMENT + "; " +
+                      DISPOSITION_PARAM_FILENAME + "=" + escapeContentDispositionFilename(write.getUri()) + contentDispositionTemporal;
+        }
+        else {
+            try {
+                disposition = DISPOSITION_TYPE_ATTACHMENT + "; " +
+                          DISPOSITION_PARAM_FILENAME + "*=UTF-8''" + URLEncoder.encode(write.getUri(), "UTF-8") + contentDispositionTemporal;
+              } catch (Exception ex) {
+                  throw new IllegalArgumentException("Uri cannot be accepted");
```

Comment:
Would the error be more useful if more specific?  Maybe "Uri cannot be encoded as UFT-8"?